cmd/build/v2/build.go

```diff
@@ -218,125 +270,80 @@ func (ob *OktetoBuilder) Build(ctx context.Context, options *types.BuildOptions)
 		}
 	}(buildsAnalytics)
 
-	ob.ioCtrl.Logger().Infof("Images to build: [%s]", strings.Join(toBuildSvcs, ", "))
-	for len(builtImagesControl) != len(toBuildSvcs) {
-		for _, svcToBuild := range toBuildSvcs {
-			// This verifies if the image was already built in this iteration, not checking if it is in the "cache" (global registry)
-			if skipServiceBuild(svcToBuild, builtImagesControl) {
-				ob.ioCtrl.Logger().Infof("skipping image '%s' due to being already built", svcToBuild)
-				continue
-			}
-			if !areAllServicesBuilt(buildManifest[svcToBuild].DependsOn, builtImagesControl) {
-				ob.ioCtrl.Logger().Infof("image '%s' can't be deployed because at least one of its dependent images(%s) are not built", svcToBuild, strings.Join(buildManifest[svcToBuild].DependsOn, ", "))
-				continue
-			}
-			if options.EnableStages {
-				ob.ioCtrl.SetStage(fmt.Sprintf("Building service %s", svcToBuild))
-			}
-
-			sp := ob.ioCtrl.Out().Spinner(fmt.Sprintf("Checking if image '%s' is already built...", svcToBuild))
-			sp.Start()
-			defer sp.Stop()
-
-			buildSvcInfo := buildManifest[svcToBuild]
-
-			// create the meta pointer and append it to the analytics slice
-			meta := analytics.NewImageBuildMetadata()
-			buildsAnalytics = append(buildsAnalytics, meta)
-
-			meta.Name = svcToBuild
-			meta.Namespace = ob.oktetoContext.GetNamespace()
-			meta.DevenvName = options.Manifest.Name
-			meta.RepoURL = ob.Config.GetAnonymizedRepo()
+	bg := newBuildGraph(buildManifest, toBuildSvcs)
+	tree, err := bg.GetGraph()
+	if err != nil {
+		return err
+	}
+	tree, err = tree.Subtree(toBuildSvcs...)
+	if err != nil {
+		return err
+	}
+	toBuildSvcs = tree.Ordered()
 
-			repoHashDurationStart := time.Now()
+	ob.ioCtrl.Logger().Infof("Images to build: [%s]", strings.Join(toBuildSvcs, ", "))
```

Comment:
Nit: This might be a bit misleading, as we don't know yet which images are going to be built. These are the potential images to check